munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala

```diff
@@ -34,21 +35,37 @@ trait ScalaCheckSuite extends FunSuite {
 
   protected def scalaCheckPrettyParameters = Pretty.defaultParams
 
+  // Allow specifying an initial seed in its Base64 encoded form
+  implicit class XtensionScalaCheckTestParameters(
+      params: ScalaCheckTest.Parameters
+  ) {
+    def withInitialSeed(encodedSeed: String): ScalaCheckTest.Parameters =
+      params.withInitialSeed(Seed.fromBase64(encodedSeed).get)
+  }
+
   private val scalaCheckPropTransform: TestTransform =
     new TestTransform("ScalaCheck Prop", t => {
       t.withBodyMap[TestValue](
         _.transformCompat {
-          case Success(prop: Prop) => propToTry(prop, t.location)
+          case Success(prop: Prop) => propToTry(prop, t)
           case r                   => r
         }(munitExecutionContext)
       )
     })
 
-  private def propToTry(prop: Prop, testLocation: Location): Try[Unit] = {
+  private def propToTry(prop: Prop, test: Test): Try[Unit] = {
     import ScalaCheckTest._
-    val result = check(scalaCheckTestParameters, prop)
-    def renderResult(r: Result) =
-      Pretty.pretty(r, scalaCheckPrettyParameters)
+    val seed = scalaCheckTestParameters.initialSeed.getOrElse(Seed.random())
+    val result = check(scalaCheckTestParameters, prop.useSeed(test.name, seed))
```

Comment:
The ScalaCheck runner uses `prop.viewSeed(name, seed)` but that causes the prop to use an hardcoded `println` to output the seed, which would look quite messy since it appears before the test failure in the output.

Instead, we determine the seed in the same way ScalaCheck does (getting it from the parameters or falling back to a random one) and inject it in the property using `useSeed` (which is what `propertyWithSeed` does in ScalaCheck)

Now we're free to print the seed ourselves, which looks much better and more consistent with the rest of the failure message.